### PR TITLE
Admin page for parents

### PR DESCRIPTION
**Pull request by sophiekrag**

### CODE REVIEW

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [inline src/pages/account/AdminPage.js]: Why not simply use `childData.stories` instead having two times the response?

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [inline src/pages/account/AdminPage.js]: Shouldn't you have some sort of response to check if it deleted the post correctly?

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [inline src/pages/account/AdminPage.js]: Why do you call this variable `response`? Response of what? Give it a more descriptive name of what it does as you're are doing multiple calls

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [reply src/pages/account/AdminPage.js]: With the response you could check for example the status code if it was deleted successfully.
If the status code is good, you can `get` the child data.

e.g:

```javascript
const deleteStory = await axiosApi.post(`deleteStory/${id}`, { childId });
const responseDelete = await deleteStory.data;

if (responseDelete.status === 200) {
  const getChildData = await axiosApi(`/getChild/${childId}`);
  const childStoryData = await getChildData.data.stories;
  setStories(childStoryData);
}
```
_(notice I'm being very specific with the variable names)_

Like this you only get the child data **IF**  the data gets deleted successfully.
Your current setup is _always_ fetching the child data regardlessly

Comment (michaelkoek) [reply src/pages/account/AdminPage.js]: Obviously, if something goes wrong you should also have it in an error message so also wrap it around a `try/catch` block. Especially because you're doing multiple requests

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [inline src/pages/account/AdminPage.js]: You're asking the user for an action with this question, so maybe instead of `alert()` the function `confirm()` would be better?

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [inline src/pages/account/AdminPage.js]: This doesn't necessarily need to be in a form.
- You don't have any inputs
- You don't have any form elements

So a form is not necessary here. You can use trigger the function with the button. In that case remove `type=submit`, because `type=submit` is only for the form.


You can pass the id directly to the `deleteStory` function, no need to store it in the `setId` function

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [inline src/pages/account/AdminPage.js]: `id` for a `useState` variable name is not a good name. Id's are usually something generic as it's a term that is used in a lot of places. So be more specific what kind of id you're stating here

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [reply src/pages/account/AdminPage.js]: I think, in that case you can even remove that state

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [inline src/pages/account/AdminPage.js]: Apparently you need `<Navbar>` in all cases. So what can you do to apply the DRY principle here?

**Review by michaelkoek — CHANGES_REQUESTED** [review]

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [reply src/pages/account/AdminPage.js]: I suggest you call the delete function directly from here and pass the ID.
Also, better to call the `confirm()`/`alert()` in the delete function to keep your `onClick` nice and clean

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [inline src/pages/account/AdminPage.js:80]: Are all these flexbox properties needed as you already are specifying the width and have a `margin: 0 auto`.
These two last properties already centers the elements

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [inline src/pages/account/AdminPage.js:21]: Why do you need here a `return`?

File: src/pages/account/AdminPage.js
Comment (sophiekrag) [reply src/pages/account/AdminPage.js]: Only private routes have a navbar. I think I can change it but I will do it when this is merged, else I think I'll get too much merge conflict right now...

Comment (michaelkoek) [reply src/pages/account/AdminPage.js]: On line 40 you also have a `<Navbar>` component.
Think about a way where you only need to use that component one for both situations

File: src/pages/account/AdminPage.js
Comment (sophiekrag) [reply src/pages/account/AdminPage.js:80]: Styling is going to be different after merge. But the only having the last two doesn't center it somehow

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [reply src/pages/account/AdminPage.js]: p.s. Don't worry about Merge conflicts. I'm actually glad that you have them. As it's also something you should learn how to resolve them

File: src/pages/account/AdminPage.js
Comment (sophiekrag) [reply src/pages/account/AdminPage.js:21]: without it I sometimes get an error. I don't know why yet

Comment (michaelkoek) [reply src/pages/account/AdminPage.js:21]: What kind of error? Maybe you can get information in the error message why you get an error

File: src/pages/account/AdminPage.js
Comment (sophiekrag) [reply src/pages/account/AdminPage.js]: If I don't wrap a form around it it doesn't work

Comment (michaelkoek) [reply src/pages/account/AdminPage.js]: Well, if you don't call the function in the `onClick` than it doesn't work.
Call the delete function in the `onclick` and pass the story `id` to it

File: src/pages/account/AdminPage.js
Comment (sophiekrag) [reply src/pages/account/AdminPage.js]: Yes, but I'll fix it when it's merged. Already have merge conflicts right now

File: src/pages/account/AdminPage.js
Comment (sophiekrag) [reply src/pages/account/AdminPage.js:21]: The useEffect error we were looking at the other day.

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [reply src/pages/account/AdminPage.js]: Fix it here, as this is code that you've added. This is also a good way to learn how to deal with merge conflicts

File: src/pages/account/AdminPage.js
Comment (sophiekrag) [reply src/pages/account/AdminPage.js]: <Button onClick={() => deleteStory(storyId)}>Delete</Button> This is what I did right now

Comment (michaelkoek) [reply src/pages/account/AdminPage.js]: Yea, that's not entirely it. Search how you can pass data with an `onclick`

File: src/pages/account/AdminPage.js
Comment (michaelkoek) [inline src/pages/account/AdminPage.js]: If you have removed the navbar than the `<>` is not necessary

**Review by michaelkoek — APPROVED** [review]

**Comment by michaelkoek** [comment]
🙈  Next time, await for at least 2 approvals before you merge

**Pull request merged**
